src/bootstrap/compile.rs

```diff
@@ -70,6 +70,33 @@ impl Step for Std {
 
         builder.ensure(StartupObjects { compiler, target });
 
+        if compiler.stage == 0 {
```

Comment:
Can you add a comment here for why this is a large special case?

Reply:
Done